src/tools.rs

```diff
@@ -481,7 +482,43 @@ pub(crate) fn time() -> i64 {
         .as_secs() as i64
 }
 
-/// Very simple email address wrapper.
+/// Struct containing all mailto information
+#[derive(Debug, Default, Eq, PartialEq)]
+pub struct MailTo {
+    pub to: Vec<EmailAddress>,
+    pub subject: Option<String>,
+    pub body: Option<String>,
+}
+
+/// Parse mailto urls
+pub fn parse_mailto(mailto_url: &str) -> Option<MailTo> {
+    if let Ok(url) = Url::parse(mailto_url) {
```

Comment:
minor note: i'd prefer to use
```
let Ok(url) = ... else {
    return None;
};
...
```
pattern to decrease code indentation

Reply:
True, that is actually pretty nice in some cases. For this I don't really think it's needed but I will keep it in the back of my head!